source/features/restore-file.tsx

```diff
@@ -86,47 +132,95 @@ async function discardChanges(progress: (message: string) => void, originalFileN
 }
 
 async function handleClick(event: DelegateEvent<MouseEvent, HTMLButtonElement>): Promise<void> {
+	event.preventDefault();
+
 	const menuItem = event.delegateTarget;
+	const filenames = getFilenames(menuItem);
 
-	const [originalFileName, newFileName = originalFileName] = menuItem
-		.closest('[data-path]')!
-		.querySelector('.Link--primary')!
-		.textContent
-		.split(' → ');
+	const commitTitle = prompt('Are you sure you want to discard these changes? Enter the commit title', `Discard changes to ${filenames.original}`);
 
-	const commitTitle = prompt(`Are you sure you want to discard the changes to ${newFileName}? Enter the commit title`, `Discard changes to ${newFileName}`);
 	if (!commitTitle) {
 		return;
 	}
 
-	await showToast(async progress => discardChanges(progress, originalFileName, newFileName, commitTitle), {
+	await showToast(async progress => discardChanges(progress, filenames.original, filenames.new, commitTitle), {
 		message: 'Loading info…',
 		doneMessage: 'Changes discarded',
 	});
 
 	// Hide file from view
-	menuItem.closest('.file')!.remove();
+	// TODO: Drop support for old view in June 2026
+	if (menuItem.tagName === 'A') {
+		menuItem.closest('.file')!.remove();
+		return;
+	}
+
+	const filesWrapper = $('div[class^="prc-PageLayout-Content-"] div[data-hpc="true"]');
+	const fileElement = [...filesWrapper.children].find(child => {
+		const filenameElement = child.querySelector('h3 code');
+		if (!filenameElement?.textContent) {
+			return false;
+		}
+
+		let filename = filenameElement.textContent
+			.replaceAll(/[\u200E\u200F\uFEFF]/g, '')
+			.trim();
+
+		// Handle rename heading "old renamed to new"
+		if (/\brenamed\s+to\b/i.test(filename)) {
+			filename = filename.split(/\s+renamed\s+to\s+/i)[0].trim();
+		}
+
+		return filename === filenames.original;
+	}) as HTMLElement;
```

Comment:
We already have a `handleMenuOpening` listener that can be used to track the last clicked file so we don't have to look for it again. Try saving it to a global on click, e.g. 

```js
let focusedFile: HTMLElement;

function handleMenuOpening(event): void {
  focusedFile = event.target.find the container that we want to .remove();
  
  etc
}
```